sdk/src/types/block/semantic/mod.rs

```diff
@@ -404,7 +404,9 @@ impl<'a> SemanticValidationContext<'a> {
 
         if self.input_mana != self.output_mana {
             if self.input_mana > self.output_mana {
-                if !self.transaction.has_capability(TransactionCapabilityFlag::BurnMana) {
+                if (self.mana_rewards.is_some() || self.reward_context_inputs.is_empty())
+                    && !self.transaction.has_capability(TransactionCapabilityFlag::BurnMana)
+                {
                     return Err(TransactionFailureReason::CapabilitiesManaBurningNotAllowed);
                 }
             } else {
```

Comment:
Don't we also have to not return the error in the else if None was provided?

Reply:
no because it's possible you aren't burning mana there

Reply:
But if the tx claims rewards which aren't provided here, there would also be a mismatch?

Reply:
yes, that's the else there

Reply:
Yeah and my point is, if `self.mana_rewards.is_none()` it shouldn't error then?

Reply:
Having a test with a transaction that claims mana would probably be the best to see if it really works

Reply:
I'm talking about this case `self.input_mana < self.output_mana`, where it now would return an error in L413?

Reply:
input_mana being smaller because the mana rewards aren't known in the validation because none was provided, but was added to the outputs

Reply:
Providing rewards and none both seem to fail
```
diff --git a/sdk/tests/client/input_selection/delegation_outputs.rs b/sdk/tests/client/input_selection/delegation_outputs.rs
index ad48472b2..bdf4e70a6 100644
--- a/sdk/tests/client/input_selection/delegation_outputs.rs
+++ b/sdk/tests/client/input_selection/delegation_outputs.rs
@@ -105,4 +105,21 @@ fn remainder_needed_for_mana() {
                 .sum::<u64>(),
         selected.transaction.outputs().iter().map(|o| o.mana()).sum::<u64>()
     );
+
+    let inputs = inputs
+        .iter()
+        .map(|input| (input.output_id(), &input.output))
+        .collect::<Vec<(&OutputId, &iota_sdk::types::block::output::Output)>>();
+
+    iota_sdk::types::block::semantic::SemanticValidationContext::new(
+        &selected.transaction,
+        &inputs,
+        None,
+        // Providing the rewards and providing None both fails
+        // None,
+        Some(std::collections::BTreeMap::from([(delegation_output_id, mana_rewards)])),
+        protocol_parameters.clone(),
+    )
+    .validate()
+    .unwrap();
 }

```

Reply:
Ah, I got input and output mana mixed up 😵‍💫

Reply:
The None case works now, but `Some(std::collections::BTreeMap::from([(delegation_output_id, mana_rewards)])),` still fails